source/Mlos.Model.Services.Client/OptimizerServiceEncoderDecoder.cs

```diff
@@ -0,0 +1,419 @@
+// -----------------------------------------------------------------------
+// <copyright file="OptimizerServiceEncoderDecoder.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root
+// for license information.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using Mlos.Model.Services.Spaces;
+
+namespace Mlos.Model.Services.Client
+{
+    public class OptimizerServiceEncoder

```

Comment:
Why not make this class static?

Reply:
I made it non-static to follow the OOP & factory model. I also prefer to define instances in order to rename something rather than using the "using x=y.z.k" model. However given that the python side follows the static-class method, perhaps that is the way to go

Reply:
Can we have a static class serve as a factory? Is there any state this object maintains?

Reply:
No state is maintained. I think you're right about this. See: [zaporter/grpc a138569c] Made Encoder&Decoder static